MigrationV3V4/CSharp/Skills/V3EchoBot/Authentication/CustomAllowedCallersClaimsValidator.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using Microsoft.Bot.Connector.SkillAuthentication;
+using System;
+using System.Collections.Generic;
+using System.Security.Claims;
+using System.Threading.Tasks;
+
+namespace Microsoft.Bot.Sample.EchoBot.Authentication
+{
+    /// <summary>
+    /// Sample claims validator that loads an allowed list from configuration if present
+    /// and checks that requests are coming from allowed parent bots.
+    /// </summary>
+    public class CustomAllowedCallersClaimsValidator : ClaimsValidator
+    {
+        private readonly IList<string> _allowedCallers;
+
+        public CustomAllowedCallersClaimsValidator(IList<string> allowedCallers)
+        {
+            // AllowedCallers is the setting in web.config file
+            // that consists of the list of parent bot ids that are allowed to access the skill
```

Comment:
In general, comments should be complete sentences and use correct punctuation and grammar. I'm not going to copy edit all the comments in this or the other code files, as many comments have this issue. I also won't hold up release of this sample because of it.
```suggestion
            // that consists of the list of parent bot IDs that are allowed to access the skill.
```
> [!NOTE]
> If you fix the comments after the related how-to article is written, try not to move lines around, or if you do let me know so I can fix the broken citations.